primitives/state-machine/src/ext.rs

```diff
@@ -757,28 +758,34 @@ where
 		let mut delete_count: u32 = 0;
 		let mut loop_count: u32 = 0;
 		let mut maybe_next_key = None;
-		self.backend
-			.apply_to_keys_while(maybe_child, maybe_prefix, maybe_cursor, |key| {
-				if maybe_limit.map_or(false, |limit| loop_count == limit) {
-					maybe_next_key = Some(key.to_vec());
-					return false
-				}
-				let overlay = match maybe_child {
-					Some(child_info) => self.overlay.child_storage(child_info, key),
-					None => self.overlay.storage(key),
-				};
-				if !matches!(overlay, Some(None)) {
-					// not pending deletion from the backend - delete it.
-					if let Some(child_info) = maybe_child {
-						self.overlay.set_child_storage(child_info, key.to_vec(), None);
-					} else {
-						self.overlay.set_storage(key.to_vec(), None);
+		let result =
+			self.backend
+				.apply_to_keys_while(maybe_child, maybe_prefix, maybe_cursor, |key| {
+					if maybe_limit.map_or(false, |limit| loop_count == limit) {
+						maybe_next_key = Some(key.to_vec());
+						return false
 					}
-					delete_count = delete_count.saturating_add(1);
-				}
-				loop_count = loop_count.saturating_add(1);
-				true
-			});
+					let overlay = match maybe_child {
+						Some(child_info) => self.overlay.child_storage(child_info, key),
+						None => self.overlay.storage(key),
+					};
+					if !matches!(overlay, Some(None)) {
+						// not pending deletion from the backend - delete it.
+						if let Some(child_info) = maybe_child {
+							self.overlay.set_child_storage(child_info, key.to_vec(), None);
+						} else {
+							self.overlay.set_storage(key.to_vec(), None);
+						}
+						delete_count = delete_count.saturating_add(1);
+					}
+					loop_count = loop_count.saturating_add(1);
+					true
+				});
+
+		if let Err(error) = result {
+			log::debug!(target: "trie", "Error while iterating the storage: {}", error);
```

Comment:
Could be good to add a comment or insist on the fact that ignoring error is a host function related behavior. Don't really know how to word it.